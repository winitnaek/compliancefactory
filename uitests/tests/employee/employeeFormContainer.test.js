import React from 'react';
import  ContactForm,{formName}  from '../../../src/app/employee/EmployeeFormContainer';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe("Employee Entry Form", () => {
  it("form name should be contact", () => {
     expect(formName).toBe('entry-form');
   });
 });
 const mockStore = configureMockStore()
 const store = mockStore()
 describe('Form component renders the form correctly renders correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Provider store={store}>
      <ContactForm/>
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});  