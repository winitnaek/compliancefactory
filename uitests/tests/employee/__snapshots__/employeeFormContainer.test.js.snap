// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders the form correctly renders correctly renders correctly 1`] = `
<div
  className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1"
>
  <h1>
    Add Employee
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        First Name
      </label>
      <br />
      <input
        className="form-control"
        name="first_name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Last Name
      </label>
      <br />
      <input
        className="form-control"
        name="last_name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        email
      </label>
      <br />
      <input
        className="form-control"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="email"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Add Employee
    </button>
  </form>
</div>
`;
